openmdao/visualization/n2_viewer/gen/NodeInfo.js

```diff
@@ -230,3 +245,86 @@ class PersistentNodeInfo extends WindowDraggable {
         }
     }
 }
+
+/**
+ * Display all connections represented by the current off-diagonal cell.
+ * @typedef NodeConnectionInfo
+ */
+class NodeConnectionInfo extends NodeInfo {
+    /**
+     * Build a list of the properties we care about and set up
+     * references to the HTML elements.
+     */
+    constructor(ui) {
+        super(ui, false);
+        this.propList = [];
+    }
+
+    /**
+     * Gather all connections between the two cells that this one represents.
+     * @param {Array} connList The connections discovered so far.
+     * @param {TreeNode} srcNode The source node to search.
+     * @param {TreeNode} tgtNode The target node to search.
+     */
+    _getConnections(connList, srcNode, tgtNode) {
+        if (srcNode.hasChildren()) {
+            for (const child of srcNode.children) {
+                this._getConnections(connList, child, tgtNode)
+            }
+        }
+        else {
+            for (const tgt of srcNode.targetParentSet) {
+                if (tgt.isLeaf() && tgt.hasParent(tgtNode, null, true)) {
+                    const newConn = { 'src': srcNode.path, 'tgt': tgt.path};
+                    connList.push(newConn)
+                }
+            }
+        }
+    }
+
+    /**
+     * Create a new persistent window by copying our contents. Set the
+     * max initial height of the window at 300px.
+     */
+    pin() {
+        if (this.tbody.html() == '') return; // Was already pinned so is empty
+
+        new PersistentNodeInfo(this, 300);
+        this.hidden = true;
+        this.clear();
+        return this;
+    }
+
+    /**
+     * Load the NodeInfo window with connection information for an off-diagonal cell.
+     * @param {Object} event Reference to the event that triggered the function.
+     * @param {MatrixCell} cell Reference to the cell that was hovered.
+     * @param {String} color The color of the title bar.
+     */
+    update(event, cell, color) {
+        if (!this.active) return;
+        this.clear();
+
+        this.name = cell.id;
+        this.ribbonColor(color);
+
+        const connList = [];
+        this._getConnections(connList, cell.srcObj, cell.tgtObj);
+        for (const conn of connList) {
+            const newRow = this.tbody.append('tr');
+
+            newRow.append('td')
+                .attr('scope', 'row')
+                .text(conn.src);
+
+            newRow.append('td').html(' &#x2501;&#x2501;&#x2501;&#x25ba; ');
+
+            newRow.append('td').text(conn.tgt);
+        }
+
+        this.title(`Connections from ${cell.srcObj.path} to ${cell.tgtObj.path}`)
```

Comment:
Is there a way to tell when srcObj and tgtObj are collapsed or not?
When things are collapsed this displays the source system and the target system.
![Connection info when collapsed](https://user-images.githubusercontent.com/699809/213254259-2f4ed6f1-13b8-4c18-95fe-d6e8835717b3.png)

But when the connection is not collapsed, this expands all the way out to the variable name:
![Connection info when not collapsed](https://user-images.githubusercontent.com/699809/213254410-c2800d5e-bc49-4bd4-988e-95a31f24a99e.png)

If it could always just show `Connections from {system} to {system}` that would be preferred.

Reply:
Also, since connections are made with promoted names, I think the variable names in each row should be promoted names instead of absolute.

Really enjoying using this!